aws_lambda_powertools/utilities/data_classes/s3_batch_operation_event.py

```diff
@@ -0,0 +1,237 @@
+import warnings
+from dataclasses import dataclass, field
+from typing import Any, Dict, Iterator, List, Optional, Tuple
+from urllib.parse import unquote_plus
+
+from aws_lambda_powertools.shared.types import Literal
+from aws_lambda_powertools.utilities.data_classes.common import DictWrapper
+
+# list of valid result code. Used both in S3BatchOperationResponse and S3BatchOperationResponseRecord
+VALID_RESULT_CODE_TYPES: Tuple[str, str, str] = ("Succeeded", "TemporaryFailure", "PermanentFailure")
+
+
+@dataclass(repr=False, order=False)
+class S3BatchOperationResponseRecord:
+    task_id: str
+    result_code: Literal["Succeeded", "TemporaryFailure", "PermanentFailure"]
+    result_string: Optional[str] = None
+
+    def asdict(self) -> Dict[str, Any]:
+        if self.result_code not in VALID_RESULT_CODE_TYPES:
+            warnings.warn(
+                stacklevel=2,
+                message=f"The resultCode {self.result_code} is not valid. "
+                "Choose from 'Ok', 'Dropped' or 'ProcessingFailed '",
```

Comment:
I don't think this list is consistent with the test on line 20. Should it be "Succeeded", "TemporaryFailure" or "PermanentFailure" ?